dali/operators/image/paste/multipaste.h

```diff
@@ -0,0 +1,217 @@
+
+  explicit MultiPasteOp(const OpSpec &spec)
+      : Operator<Backend>(spec)
+      , output_type_arg_(spec.GetArgument<DALIDataType>("dtype"))
+      , output_type_(DALI_NO_TYPE)
+      , input_type_(DALI_NO_TYPE)
+      , output_size_("output_size", spec)
+      , in_idx_("in_ids", spec)
+      , in_anchors_("in_anchors", spec)
+      , in_shapes_("shapes", spec)
+      , out_anchors_("out_anchors", spec) {
+    zero_anchors_ = Coords(zeros_, {2});
+
+    if (std::is_same<Backend, GPUBackend>::value) {
+      kernel_manager_.Resize(1, 1);
+    } else {
+      kernel_manager_.Resize(num_threads_, max_batch_size_);
+    }
+  }
+
+  bool CanInferOutputs() const override {
+    return true;
+  }
+
+  template<int ndim = 2>
+  bool Intersects(const Coords& anchors1, const Coords& shapes1,
+                  const Coords& anchors2, const Coords& shapes2) const {
+    for (int i = 0; i < ndim; i++) {
+      if (anchors1.data[i] + shapes1.data[i] <= anchors2.data[i]
+          || anchors2.data[i] + shapes2.data[i] <= anchors1.data[i]) {
+        return false;
+      }
+    }
+    return true;
+  }
+
+  void AcquireArguments(const OpSpec &spec, const workspace_t<Backend> &ws) {
+    const auto &images = ws.template InputRef<Backend>(0);
+
+    auto curr_batch_size = ws.GetRequestedBatchSize(0);
+    if (curr_batch_size == 0)
+      return;
+
+    output_size_.Acquire(spec, ws, curr_batch_size, true);
+    in_idx_.Acquire(spec, ws, curr_batch_size, false);
+    if (out_anchors_.IsDefined()) {
+      out_anchors_.Acquire(spec, ws, curr_batch_size, false);
+    }
+    if (in_anchors_.IsDefined()) {
+      in_anchors_.Acquire(spec, ws, curr_batch_size, false);
+    }
+    if (in_shapes_.IsDefined()) {
+      in_shapes_.Acquire(spec, ws, curr_batch_size, false);
+    }
+    input_type_ = ws.template InputRef<Backend>(0).type().id();
+    output_type_ =
+        output_type_arg_ != DALI_NO_TYPE
+        ? output_type_arg_
+        : input_type_;
+
+    raw_input_size_mem_.clear();
+    const auto &input_shape = images.shape();
+    const int spatial_ndim = 2;  // in the future we may want to support 3D
+    raw_input_size_mem_.reserve(spatial_ndim * input_shape.num_samples());
+    for (int i = 0; i < curr_batch_size; i++) {
+      auto shape = input_shape.tensor_shape_span(i);
+      for (int j = 0; j < spatial_ndim; j++) {
+        auto extent = static_cast<int>(shape[j]);
+        raw_input_size_mem_.push_back(extent);
+      }
+    }
+
+    no_intersections_.clear();
+    for (int i = 0; i < curr_batch_size; i++) {
+      const int64_t n_paste = in_idx_[i].shape[0];
+
+      if (in_anchors_.IsDefined()) {
+        DALI_ENFORCE(in_anchors_[i].shape[0] == n_paste,
+                     "in_anchors must be same length as in_idx");
+        DALI_ENFORCE(in_anchors_[i].shape[1] == spatial_ndim,
+         "in_anchors must have number of coordinates equal to that of input images - 1 (channel)");
```

Comment:
Just a suggestion, I think it'd be better for the user to have something like:
```suggestion
         make_string("Unexpected number of dimensions for ``in_anchors``. Expected ", spatial_dim, ", got ", in_anchors_[i].shape[1]));
```